cli/deno_dir.rs

```diff
@@ -32,8 +28,37 @@ use std::sync::Mutex;
 use url;
 use url::Url;
 
-const SUPPORTED_URL_SCHEMES: [&str; 3] = ["http", "https", "file"];
+#[derive(Debug, Clone)]
+pub struct SourceFile {
+  pub url: Url,
+  pub redirect_source_url: Option<Url>,
+  pub filename: PathBuf,
+  pub media_type: msg::MediaType,
+  pub source_code: Vec<u8>,
+}
+
+impl SourceFile {
+  // TODO: this method should be implemented on CompiledSourceFile trait
```

Comment:
I don't see CompiledSourceFile anywhere here. Is the TODO saying that a new trait called CompiledSourceFile should be created with `js_source()` method?

Reply:
More-or-less, more info in https://github.com/denoland/deno/pull/2636#discussion_r303029242